data/lua/wml-tags.lua

```diff
@@ -141,7 +141,9 @@ function wml_actions.fire_event(cfg)
 	local w2 = helper.get_child(cfg, "secondary_attack")
 	if w2 then w1 = w1 or {} end
 
-	wesnoth.fire_event(cfg.name, x1, y1, x2, y2, w1, w2)
+	if cfg.id and cfg.id ~= "" then wesnoth.fire_event_by_id(cfg.id, x1, y1, x2, y2, w1, w2)
+	elseif cfg.name and cfg.name ~= "" then wesnoth.fire_event_by_name(cfg.name, x1, y1, x2, y2, w1, w2)
+	end
```

Comment:
Thinking about it, what should happen if both id and name are given? (By the way, is it possible to have an event without a name?) I can see two possibilities: id takes precedence (and a warning is printed), or the event only fires if both id and name match.